photon-core/src/main/java/org/photonvision/vision/aruco/PhotonArucoDetector.java

```diff
@@ -17,116 +17,90 @@
-                        cornerMat.get(0, 3)[0]
-                    };
-            yCorners =
-                    new double[] {
-                        cornerMat.get(0, 0)[1],
-                        cornerMat.get(0, 1)[1],
-                        cornerMat.get(0, 2)[1],
-                        cornerMat.get(0, 3)[1]
-                    };
-            cornerMat.release();
+    public void setParams(DetectorParameters params) {
+        detector.setDetectorParameters(params);
+    }
 
-            double[] tvec;
-            double[] rvec;
-            if (coeffs != null) {
-                // Need to apply a 180 rotation about Z
-                var origRvec = rvecs.get(i, 0);
-                var axisangle = VecBuilder.fill(origRvec[0], origRvec[1], origRvec[2]);
-                Rotation3d rotation = new Rotation3d(axisangle, axisangle.normF());
-                var ocvRotation = ARUCO_BASE_ROTATION.rotateBy(rotation);
+    /**
+     * Detect fiducial tags in the grayscaled image using the {@link ArucoDetector} in this class.
+     * Parameters for detection can be modified with {@link #setParams(DetectorParameters)}.
+     *
+     * @param grayscaleImg A grayscaled image
+     * @return An array of ArucoDetectionResult, which contain tag corners and id.
+     */
+    public ArucoDetectionResult[] detect(Mat grayscaleImg) {
+        // detect tags
+        detector.detectMarkers(grayscaleImg, cornerMats, ids);
 
-                var angle = ocvRotation.getAngle();
-                var finalAxisAngle = ocvRotation.getAxis().times(angle);
+        ArucoDetectionResult[] toReturn = new ArucoDetectionResult[cornerMats.size()];
+        for (int i = 0; i < cornerMats.size(); i++) {
+            // each detection has a Mat of corners
+            cornerMat = cornerMats.get(i);
 
-                tvec = tvecs.get(i, 0);
-                rvec = finalAxisAngle.getData();
-            } else {
-                tvec = new double[] {0, 0, 0};
-                rvec = new double[] {0, 0, 0};
-            }
+            // Aruco detection returns corners (BR, BL, TL, TR).
+            // For parity with AprilTags and photonlib, we want (BL, BR, TR, TL).
+            double[] xCorners = {
+                cornerMat.get(0, 1)[0],
+                cornerMat.get(0, 0)[0],
+                cornerMat.get(0, 3)[0],
+                cornerMat.get(0, 2)[0]
+            };
+            double[] yCorners = {
+                cornerMat.get(0, 1)[1],
+                cornerMat.get(0, 0)[1],
+                cornerMat.get(0, 3)[1],
+                cornerMat.get(0, 2)[1]
+            };
+            cornerMat.release();
 
-            toReturn[i] =
-                    new ArucoDetectionResult(xCorners, yCorners, (int) ids.get(i, 0)[0], tvec, rvec);
+            toReturn[i] = new ArucoDetectionResult(xCorners, yCorners, (int) ids.get(i, 0)[0]);
         }
-        rvecs.release();
-        tvecs.release();
+
         ids.release();
 
+        // sort tags by ID
```

Comment:
Nit: there's really no way in the std lib to sort a stream without having to manually return 1/-1/0 yourself?